pkg/sdk/testint/basic_object_tracking_integration_test.go

```diff
@@ -0,0 +1,100 @@
+package testint
+
+import (
+	"context"
+	"fmt"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/snowflakedb/gosnowflake"
+	"github.com/stretchr/testify/require"
+	"strings"
+	"testing"
+)
+
+// Research for basic object tracking done as part of SNOW-1737787
+
+// https://docs.snowflake.com/en/sql-reference/parameters#query-tag
+func TestInt_ContextQueryTags(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	sessionId, err := client.ContextFunctions.CurrentSession(ctx)
+	require.NoError(t, err)
+
+	queryTag := "some query tag"
+	require.NoError(t, client.Sessions.AlterSession(ctx, &sdk.AlterSessionOptions{
+		Set: &sdk.SessionSet{
+			SessionParameters: &sdk.SessionParameters{
+				QueryTag: sdk.String(queryTag),
+			},
+		},
+	}))
+	t.Cleanup(func() {
+		_, err = client.QueryUnsafe(ctx, "ALTER SESSION UNSET QUERY_TAG")
```

Comment:
Why not Sessions.AlterSession with unset?

Reply:
done